plugins/readtimer.koplugin/main.lua

```diff
@@ -0,0 +1,95 @@
+
+local Font = require("ui/font")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local T = require("ffi/util").template
+local _ = require("gettext")
+
+local ReadTimer = WidgetContainer:new{
+    name = "readtimer",
+    time = 0,  -- The expected time of alarm if enabled, or 0.
+}
+
+function ReadTimer:init()
+    self.alarm_callback = function()
+        if self.time == 0 then return end -- How could this happen?
+        self.time = 0
+        UIManager:show(InfoMessage:new{
+            text = T(_("Read timer alarm\nTime's up. It's %1 now."), os.date("%c")),
+            timeout = 10,
+        })
+    end
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function ReadTimer:scheduled()
+    return self.time ~= 0
+end
+
+function ReadTimer:remainingMinutes()
+    if self:scheduled() then
+        return os.difftime(self.time, os.time()) / 60
+    else
+        return math.huge
+    end
+end
+
+function ReadTimer:unschedule()
+    if self:scheduled() then
+        UIManager:unschedule(self.alarm_callback)
+        self.time = 0
+    end
+end
+
+function ReadTimer:addToMainMenu(tab_item_table)
+    table.insert(tab_item_table.plugins, {
+        text = _("Read timer"),
+        callback = function()
+            local title = _("When will the countdown timer alarm?\nThe unit is \"minute\", and only positive number is accepted.")
+            if self:scheduled() then
+                title = title .. T(_("\nYou have already set up a timer in %1 minutes. Setting a new one will overwrite it."),
+                                   string.format("%.2f", self:remainingMinutes()))
+            end
+            local buttons = {{
+                text = _("Close"),
+                callback = function()
+                    UIManager:close(self.input)
+                end,
+            }, {
+                text = _("Start"),
+                callback = function()
+                    self:unschedule()
+                    local seconds = self.input:getInputValue() * 60
+                    if seconds > 0 then
+                        self.time = os.time() + seconds
+                        UIManager:scheduleIn(seconds, self.alarm_callback)
+                    end
+                    UIManager:close(self.input)
+                end,
+            }}
+            if self:scheduled() then
+                table.insert(buttons, {
+                    text = _("Stop"),
+                    callback = function()
+                        self:unschedule()
+                        UIManager:close(self.input)
+                    end,
+                })
+            end
+            self.input = InputDialog:new{
+                title_face = Font:getFace("cfont", 20),
```

Comment:
I have a suggestion, how about we adding a "description" field for InputDialog? If it is set, we can use a TextBoxWidget to display it under the title. It looks like a very common scenario, since the information is usually not enough by using only title and input hint.

I have updated the change, you may have a look once I pushed.